<!-- JHCap Camera Launch File -->
<launch>
  <!-- Camera node arguments -->
  <arg name="camera_name" default="jhcap_camera" />
  <arg name="camera_id" default="0" />
  <arg name="frame_id" default="camera" />
  <arg name="image_width" default="1920" />
  <arg name="image_height" default="1080" />
  <arg name="frame_rate" default="55.0" />
  <arg name="gain" default="174" />
  <arg name="exposure" default="1000" />
  <arg name="auto_exposure" default="false" />
  <arg name="auto_gain" default="false" />
  <arg name="jpeg_quality" default="100" />
  
  <!-- Camera info file -->
  <arg name="camera_info_url" default="package://jhcap_camera/config/camera_info.yaml" />
  
  <!-- Debug options -->
  <arg name="debug" default="false" />
  <arg name="launch_prefix" default="" />
  <arg name="respawn" default="false" />
  
  <!-- Node namespace -->
  <group ns="$(arg camera_name)">
    <!-- Camera node -->
    <node name="jhcap_camera" 
          pkg="jhcap_camera" 
          type="jhcap_camera" 
          output="screen"
          launch-prefix="$(arg launch_prefix)"
          respawn="$(arg respawn)">
      
      <!-- Camera parameters -->
      <param name="camera_name" value="$(arg camera_name)" />
      <param name="camera_id" value="$(arg camera_id)" />
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="image_width" value="$(arg image_width)" />
      <param name="image_height" value="$(arg image_height)" />
      <param name="frame_rate" value="$(arg frame_rate)" />
      <param name="gain" value="$(arg gain)" />
      <param name="exposure" value="$(arg exposure)" />
      <param name="auto_exposure" value="$(arg auto_exposure)" />
      <param name="auto_gain" value="$(arg auto_gain)" />
      <param name="jpeg_quality" value="$(arg jpeg_quality)" />
      
      <!-- Camera info -->
      <param name="camera_info_url" value="$(arg camera_info_url)" />
      
      <!-- Remap topics if needed -->
      <remap from="image_raw" to="image_raw" />
      <remap from="image_raw/compressed" to="image_raw/compressed" />
      <remap from="camera_info" to="camera_info" />
    </node>
    
    <!-- Image processing nodes (optional) -->
    <!-- Uncomment if you want image processing -->
    <!--
    <include file="$(find image_proc)/launch/image_proc.launch">
      <arg name="manager" value="camera_nodelet_manager" />
    </include>
    -->
  </group>
  
  <!-- Optional: Start image view for debugging 
  <node name="image_view" 
        pkg="image_view" 
        type="image_view" 
        output="screen"
        if="$(arg debug)">
    <remap from="image" to="/$(arg camera_name)/image_raw" />
  </node> -->
  
  <!-- Optional: Start rqt_image_view for debugging 
  <node name="rqt_image_view" 
        pkg="rqt_image_view" 
        type="rqt_image_view" 
        output="screen"
        if="$(arg debug)">
  </node> -->
</launch>